tests/snappi_tests/intf_utils/intf_accept_metrics.py

```diff
@@ -0,0 +1,109 @@
+# This file defines the interfaces that snappi tests accept external metrics.
+
+#  Metrics data are organized into the hierarchies below
+#  ResourceMetrics
+#    ├── ResourceID
+#    └── ScopeMetrics
+#       ├── ScopeID
+#       └── Metric
+#          ├── Name
+#          ├── Description
+#          ├── Unit
+#          ├── metadata
+#          └── data
+#             └── Gauge
+#
+# A ResourceMetrics has its ID and a list of ScopeMetrics objects.
+# A ScopeMetrics    has its ID and a list of Metric objects.
+# A Metric has several attributes and data. So far we only have Gauge type data.
+# A Gauge has a list of NumberDataPoint objects.
+# A NumberDataPoint has its label, value, flags and the timestamp at which the data was collected.
+#                         +---------------------+
+#                         | DataPoint 1         |
+#                         | +---------+ +-----+ |
+#               +-----+   | |timestamp| |label| |
+#               |  1  |-->| +---------+ +-----+ |
+#               +-----+   |                     |
+#               |  .  |   | +-----+ +-----+     |
+#               |  .  |   | |value| |flags|     |
+#               |  .  |   | +-----+ +-----+     |
+#               |  .  |   +---------------------+
+#               |  .  |              .
+#               |  .  |              .
+#               |  .  |              .
+#               |  .  |   +---------------------+
+#               |  .  |   | DataPoint M         |
+#               +-----+   | +---------+ +-----+ |
+#               |  M  |-->| |timestamp| |label| |
+#               +-----+   | +---------+ +-----+ |
+#                         |                     |
+#                         | +-----+ +-----+     |
+#                         | |value| |flags|     |
+#                         | +-----+ +-----+     |
+#                         +---------------------+
```

Comment:
We need an interface to emit these metrics. To be more specific:

Factory class, so we can override:

```python
class MetricReporterFactory:
    def __init__(self, <test run related info>, <testbed related info>):
        self.xxx = xxx

    def build(self):
        # Create MetricsReporter here.
        pass
```

Interface for test to use:
```python
class MetricsReporter:
    def __init__(self, <test run related info>, <testbed related info>):
        self.xxx = xxx

    def emit_metrics(metrics: TestMetrics):
        # throw not implemented
```

Reply:
Added these classes. They only have prototype as of now. Will add the implementation later.